File: backend-spring-auth/src/main/java/com/woohuhu/spring_backend_auth/user/service/UserService.java
package com.woohuhu.spring_backend_auth.user.service;

import com.woohuhu.spring_backend_auth.user.dto.*;
import org.springframework.stereotype.Service;

@Service
public interface UserService {
    UserInfoDto getUser(String id) throws Exception;

    int createUser(UserDto userDto) throws Exception;

    UserDto authenticate(LoginRequestDto loginRequestDto) throws Exception;

    int createRefreshToken(String id, String refreshToken) throws Exception;

    int deleteRefreshToken(String id) throws Exception;
}

File: frontend/src/service/user/user.service.js
import { UserController } from "./../../controller/user.controller";

export class UserService {
  static createUser = async (req) => {
    try {
      const result = await UserController.createUser(req);
      return result.data;
    } catch (error) {
      throw new Error(error);
    }
  };

  static refreshToken = async (req)=>{
    try{
      const result = await UserController.refreshToken(req);
      return result.data;
    }catch(error){
      throw new Error(error);
    }
  }
}
